libs/ha/ha_common_libs_k8s.py

```diff
@@ -1480,3 +1483,40 @@ def delete_single_pod_setting_replica_0(self, master_node_obj, health_obj,
             return False, resp
         LOGGER.info("Services of pod are in online state")
         return True, pod_name
+
+    def mark_node_failure(self, mnode_obj, pod_list: list, go_random: bool = False,
+                          rsc_opt: str = "mark_node_failure", rsc: str = "node"):
+        """
+        Helper function to set node/pod (or multi-node/pod) status to Failed random if go_random.
+        :param pod_list: List of pod/nodes to be marked as failed
+        :param go_random: If True, send mark failure signal to node/pod randomly
+        :param mnode_obj: Master node object to fetch the pod/node ID
+        :param rsc_opt = Operation to be performed on resource (eg. mark_node_failure)
+        :param rsc = resource type (eg. node)
+        :return: bool, response
+        """
+        pod_info = {}
+        pod_data = {'id': None, 'status': False}
+        for pod in pod_list:
+            pod_info[pod] = pod_data.copy()
+            pod_info[pod]['id'] = mnode_obj.get_machine_id_for_pod(pod)
+
+        if go_random:
+            for pod in pod_list:
+                if self.system_random.choice([True, False]):
+                    data_val = {"operation": rsc_opt,
+                                "arguments": {"id": f"{pod_info[pod]['id']}"}}
+                    resp = self.system_health.post_resource_signal(req_body=data_val,
+                                                                   resource=rsc)
+                    pod_info[pod]['status'] = resp[0]
```

Comment:
resp of POST call will contain True/False depending on call status, it is not pod status.
May be you need to do get resource status here and add that status in pod_info.